File: src/main/java/app/askresume/global/error/exception/BusinessException.java
package app.askresume.global.error.exception;

import app.askresume.global.error.ErrorCode;
import lombok.Getter;

@Getter
public class BusinessException extends RuntimeException {
    private final ErrorCode errorCode;

    public BusinessException(ErrorCode errorCode) {
        super(errorCode.getProperties());
        this.errorCode = errorCode;
    }

}

File: src/main/kotlin/app/askresume/global/error/exception/BusinessException.kt
package app.askresume.global.error.exception

import app.askresume.global.error.ErrorCodes

open class BusinessException(
    open val codeBook: ErrorCodes,
    val properties: String,
    val arguments: Array<Any>,
    override val cause: Throwable? = null,
) : RuntimeException(codeBook.description, cause)